### PR TITLE
Tweaked Build 2018-12-02

### PR DESCRIPTION
**Pull request by luciano-croce**
- Concise plugin description (detailed plugin guidelines not allow more of 151 characters)
- Use of use Semantic Versioning MAJOR.MINOR.PATCH 1.0.0-beta instead 1.0-beta
- Corrected license spacing
- Network support (multisite)
- Secureds ABSPATH code
- Corrected capitalization for Block Editor and Classic Editor
- Removes redundant words to maintain description in the same line
- Corrects the case where the plugin updates lose and delete user-defined custom settings
- Corrects the case where the plugin deactivation lose and delete user-defined custom settings
- Minor general code tweaks

### CODE REVIEW

File: classic-editor.php [hunk 1]
```diff
@@ -1,30 +1,63 @@
...
+ * - Removes redundant words to maintain description in the same line
+ * - Corrects the case where the plugin updates lose and delete user-defined custom settings
+ * - Corrects the case where the plugin deactivation lose and delete user-defined custom settings
+ *
  */
 
-if ( ! defined( 'ABSPATH' ) ) {
-	die( 'Invalid request.' );
-}
+defined( 'ABSPATH' ) || die( 'Invalid request.' ); # TWEAK: alternative code more secure?
```
Comment (azaozz) [inline classic-editor.php]: Nope, not more secure. It's exactly the same just written in a bit more "obscure" way :)

File: classic-editor.php [hunk 2]
```diff
@@ -1,30 +1,63 @@
...
+
+if ( ! defined( 'PLUGIN_CLASSIC_EDITOR_BUILD' ) ) define( 'PLUGIN_CLASSIC_EDITOR_BUILD', '2018-12-02' );
+
+	add_filter( 'plugin_row_meta', 'classic_editor_adds_row_meta_build', 10, 4 ); # TWEAK: only for development version?
+
+	/**
+	 * Adds Plugin Row Meta Build and Language.
+	 */
+	function classic_editor_adds_row_meta_build( $plugin_meta, $plugin_file ) {
```
Comment (azaozz) [inline classic-editor.php]: Why do we need this? If it's useful meta, lets add it to core, or at least an uniform way for it.

Comment (luciano-croce) [reply classic-editor.php]: OK

File: classic-editor.php [hunk 3]
```diff
@@ -676,7 +735,16 @@ public static function activate() {
...
+	public static function deactivate() {               # TWEAK: prevents plugin updates or simply deactivating (but not deleting or uninstalling) losing custom user-defined settings
+//		delete_option( 'classic-editor-replace' );
+//		delete_option( 'classic-editor-remember' );
+//		delete_option( 'classic-editor-allow-users' );
+	}
+
+	/**
+	 * Delete the options on uninstallation.
+	 */
+	public static function uninstall() {                # TWEAK: removes custom user defined settings only when plugin is deleted or uninstalled
```
[the remark this answers is not in the document]
Comment (luciano-croce) [reply classic-editor.php]: Deleting the options only on uninstall prevent this accidental case ;)

[the remark this answers is not in the document]
Comment (luciano-croce) [reply classic-editor.php]: IMHO is more reliable one "admin button" RESET TO DEFAULT" for example... ;)

**Comment by luciano-croce** [comment]
Thanks @azaozz for all mergings :)

**Pull request merged**